### DISCUSSION TITLE
Dynamic DNS

### DISCUSSION THREAD

**Discussion by sisimomo** [homeassistant-apps/app-cloudflared#519] [Ideas]
It would be highly beneficial to incorporate a solution such as [cloudflare-ddns](https://github.com/timothymiller/cloudflare-ddns) into this addons. This way, users who wish to link a domain name to their Dynamic IP address can effortlessly achieve this without the need for an additional addon, like the [one I've developed](https://github.com/sisimomo/home-assistant-addons/tree/main/cloudflare_ddns) for this specific purpose. This implementation would empower users to use domain names instead of IP addresses when connecting to services that cannot pass through a tunnel (e.g., VPN, etc.).

**Comment by sisimomo** [comment]
Here a interting article from cloudflare about Dynamic DNS (DDNS).
https://www.cloudflare.com/en-ca/learning/dns/glossary/dynamic-dns/

**Comment by elcajon** [comment]
Thanks for the suggestion. I have to be honest and say that I don't think it's advantageous to mix the tunnel functionality with DDNS. The latter requires appropriate router configuration etc. which we would have to document and maintain accordingly.
Is the standard [integration](https://www.home-assistant.io/integrations/cloudflare/)  not sufficient for this use case?

**Comment by sisimomo** [comment]
I wasn't aware of this integration. You're correct; it can fulfill what I'm searching for!
